File: ViewModels/ReservationViewModel.cs
using System;
using System.Collections.Generic;
using Entrepot.Data.Entities;

namespace Entrepot.ViewModels
{
  public class ReservationViewModel{
        
       
        public DateTime DateDeb;
        public DateTime Datefin;
        public List<LigneProduit> LignesProduits { get => LignesProduit; set => LignesProduit = value; }
        List<LigneProduit> LignesProduit { get; set; } = new List<LigneProduit> ();
        public Client Client{ get; set; }
        public Utilisateur Employee{ get; set; }
    }


}

File: ClientApp/ViewModels/ReservationViewModel.ts
import { Guid } from "./Guid";
import { ClientViewModel } from "./ClientViewModel";
import { EmployeeViewModel } from "./EmployeeViewModel";

import { LigneProduitViewModel } from "./LigneProduitViewModel";


export class ReservationViewModel {
    public Id: string;
    public DateDeb : Date;
    public DateFin : Date;
    public LignesProduits: Array<LigneProduitViewModel> = new Array<LigneProduitViewModel>();
    public Client: ClientViewModel;
    public Employee : EmployeeViewModel;
}